packages/Dockerfile.debian

```diff
@@ -1,36 +1,64 @@
-ARG DOSBOX_VERS=v0.00.0
-ARG BASE_IMG=ghcr.io/dosbox-staging/debian-cross-base
-ARG DEB_VERS=10
-
-FROM ${BASE_IMG}:${DEB_VERS}-${DOSBOX_VERS} as final
-ARG DPKG_ARCH=arm64
+ARG DEB_VERS
+ARG DPKG_ARCH=""
 ARG DOSBOX_APT_LIST=/debian-apt-list.txt
+FROM debian:${DEB_VERS}-slim
 ARG DEB_VERS
+ARG DPKG_ARCH
+ARG DOSBOX_APT_LIST
+
+# Enable cross compile architecture. Abort if not set correctly
+RUN case "$DPKG_ARCH" in \
+        arm64|armhf|ppc64el) dpkg --add-architecture $DPKG_ARCH ;; \
+        *)  echo "DPKG_ARCH not set!" && exit 1 ;; \
+    esac
 
-ENV MESON_CROSS_FILE=/usr/local/share/meson/cross/dosbox-cross
+ENV DEBIAN_FRONTEND=noninteractive
+# Install common base package
+RUN apt-get update && apt-get -y install --no-install-recommends \
+        ca-certificates \
+        git \
+        ninja-build \
+        python3-pip \
+        pkg-config \
+        build-essential \
+        qemu-user-static \
+    && rm -rf /var/lib/apt/lists/*
+
+# Meson cross file location if cross compiling
+ARG MESON_CROSS_FILE=/usr/local/share/meson/cross/dosbox-cross
+
+RUN echo "Debian version is ${DEB_VERS}"
 
 COPY ./debian-${DEB_VERS}-docker-apt.txt ${DOSBOX_APT_LIST}
 
-RUN mkdir -p $(dirname ${MESON_CROSS_FILE})
-RUN apt-get update && apt-get -y install --no-install-recommends \
+# Install cross toolchain and generate meson 
+# cross file using debcrossgen
+RUN  mkdir -p $(dirname ${MESON_CROSS_FILE}) \
+    && apt-get update \
```

Comment:
This is running `apt-get update` again after we've already done it once above.  Does this need to be here?

Reply:
Notice the `rm -rf /var/lib/apt/lists/*` at the end of the command. This is standard practice when creating a docker image to minimize layer size, as each layer is immutable.

On the other hand, the debian package list probably isn't *that* large, but still.